decidim-core/app/controllers/concerns/decidim/locale_switcher.rb

```diff
@@ -17,7 +17,7 @@ module LocaleSwitcher
       def set_locale
         locale = if params["locale"]
                    params["locale"]
-                 elsif current_user && current_user.locale.present?
+                 elsif current_user&.locale.present?
```

Comment:
@mrcasals Won't we get the same problem you explained in https://github.com/decidim/decidim/pull/1945#discussion_r143203909 if `current_user` is `nil`?

Reply:
You're right, I missed it when undoing the changes :( Fixed, thanks!